packages/console/src/pages/ApplicationDetails/index.tsx

```diff
@@ -97,6 +98,23 @@ const ApplicationDetails = () => {
     toast.success(t('general.saved'));
   });
 
+  const onDelete = async () => {
+    if (!data || isDeleting) {
+      return;
+    }
+
+    try {
+      await api.delete(`/api/applications/${data.id}`);
+      setIsDeleted(true);
+      setIsDeleting(false);
+      setIsDeleteFormOpen(false);
+      toast.success(t('application_details.application_deleted', { name: data.name }));
+      navigate(`/applications`);
+    } catch {
+      setIsDeleting(false);
+    }
```

Comment:
nit: just an insight. feels like a good place to extract it as a hook or maybe that takes the delete endpoint and a callback URL as props.

Reply:
Maybe good to extract the CRUD operations & states into a hook in the future.
Leave it to LOG-3638.